src/main/python/prototype/text_extract.py

```diff
@@ -12,68 +12,52 @@
 """
 
 from PIL import Image
-from preprocessing import ThresholdingManager
-from preprocessing import BlurManager
-from preprocessing import ColorManager
-from preprocessing import SimplificationManager
-from processing import FaceDetector
-from processing import BarCodeManager
-from processing import TextManager
+from prototype.preprocessing.thresholding_manager import ThresholdingManager
+from prototype.preprocessing.blur_manager import BlurManager
+from prototype.preprocessing.color_manager import ColorManager
+from prototype.preprocessing.simplification_manager import SimplificationManager
+from prototype.processing.face_manager import FaceDetector
+from prototype.processing.barcode_manager import BarCodeManager
+from prototype.processing.text_manager import TextManager
 
 import pytesseract
 import cv2
 import os
-import json
 
 # Constants path to trained data for Shape Predictor.
 SHAPE_PREDICTOR_PATH = "{base_path}/trained_data/shape_predictor_face_landmarks.dat".format(
     base_path=os.path.abspath(os.path.dirname(__file__)))
 
 
 class TextExtractor:
-    def extract(img, thresh="adaptive", blurr="median", clr="red", rm=False, knl=[7]):
+    def extract(self, img, thresh="adaptive", blurr="median", clr="red", rm=False, knl=[9]):
 
-        image = cv2.imread(img)
+        image = img
 
         simplification_manager = SimplificationManager()
         barcode_manager = BarCodeManager()
         color_manager = ColorManager()
         face_detector = FaceDetector(SHAPE_PREDICTOR_PATH)
         image = simplification_manager.perspectiveTransformation(image)
         cv2.imwrite("output/3-warped.png", image)
-
+        data = {}
         barcode_data_found, barcode_scan_data, image = barcode_manager.get_barcode_info(image)
         if barcode_data_found:
             data = {
-                'ID_number': barcode_scan_data.decode('utf-8'),
+                'identity_number': barcode_scan_data.decode('utf-8'),
             }
-            card_data = json.dumps(data)
-            print(card_data)
 
         if rm is True:
             image = face_detector.blur_face(image)
             cv2.imwrite("output/4-faceRemvoal.png", image)
 
-        if clr is not None:
-            if clr == "blackhat":
-                image = color_manager.blackHat(image)
-            elif clr == "tophat":
-                image = color_manager.topHat(image)
-            else:
-                image = color_manager.extractChannel(image, clr)
-            cv2.imwrite("output/5-colour_extract.png", image)
-
-        cv2.imwrite("output/colour_extract.png", image)
         if knl is not None:
```

Comment:
Swap the `if` and the `else` block in order to remove the unnecessary check for `None`.
```python
if knl:
    if blurr == "median":
        blur_kernel = [9]
    else:
        blur_kernel = [(9,9)]
else:
    blur_kernel = knl
```